src/dune/context.ml

```diff
@@ -147,8 +147,29 @@ let opam_config_var ~env ~cache var =
         Some s
       | Error _ -> None
 
+let best_prog dir prog =
+  let fn = Path.relative dir (prog ^ ".opt" ^ Bin.exe) in
+  if Bin.exists fn then
+    Some fn
+  else
+    let fn = Path.relative dir (prog ^ Bin.exe) in
+    if Bin.exists fn then
+      Some fn
+    else
+      None
+
+let which ~path prog =
+  let rec search = function
+    | [] -> None
+    | dir :: rest ->
+      match best_prog dir prog with
+      | None -> search rest
+      | Some fn -> Some fn
+  in
+  search path
```

Comment:
I copied this code from `Bin`, but I guess I can use `List.find_map path ~f:(fun dir -> best_prog dir prog)` instead.

Reply:
Using `List.find_map` seems good indeed

Reply:
Done.